### ISSUE TITLE
[<Ray component: Train>] Benchmark testing on Mosaic Composer with Ray

### ISSUE THREAD

**Issue by sharlec** [ray-project/ray#32946]
### What happened + What you expected to happen

I am trying to use Mosaic Training with Ray in multiple GPUs. I followed the example from [this page](https://docs.ray.io/en/latest/train/api/doc/ray.train.mosaic.MosaicTrainer.html) 

I ran some benchmark testing and wish to observe the performance of multiple GPUs compare to single GPU for Mosaic trainer.

I use batch_size = 512 and max_duration as 1000 batches to run the test.

In case of single worker, the GPU utilization is about 20% and execution is about 70.47 seconds.
In case of 2 workers, the GPU utilization on both servers reach 100% and the execution time is about 808.41 seconds.

I do not understand this result, what make the GPU utilization even higher when split the tasks to multiple GPUs? Why it takes longer to train?

I had seen similar benchmark testing had been done in [Issue #28451](https://github.com/ray-project/ray/pull/28451), which multiple workers are supposed reduce the training time.  If possible, may I know more details about this?

### Versions / Dependencies

Ray: 2.3.0
Python: 3.8.10
OS: Ubuntu 20.04.5 LTS
MosaicML: 0.12.1
torch=1.13.1
torch-optimizer = 0.3.0
torchvision=0.14.1
transformeres=4.26.1

There are two servers in my environment, each one contains 16 CPUs and 1 GPU.

### Reproduction script

import torch.utils.data  
import torchvision  
from torchvision import transforms, datasets  
from composer.models.tasks import ComposerClassifier 
import composer.optim 
from composer.algorithms import LabelSmoothing 
import ray
from ray.air.config import ScalingConfig
import ray.train as train
from ray.air import session
from ray.train.mosaic import MosaicTrainer 

BATCH_SIZE = 512

def trainer_init_per_worker(config):
    # prepare the model for distributed training and wrap with
    # ComposerClassifier for Composer Trainer compatibility
    model = torchvision.models.resnet18(num_classes=10)
    model = ComposerClassifier(ray.train.torch.prepare_model(model))
    # prepare train/test dataset
    mean = (0.507, 0.487, 0.441)
    std = (0.267, 0.256, 0.276)
    cifar10_transforms = transforms.Compose(
        [transforms.ToTensor(), transforms.Normalize(mean, std)]
    )
    data_directory = "~/data"
    train_dataset = datasets.CIFAR10(
        data_directory,
        train=True,
        download=True,
        transform=cifar10_transforms
    )
    # prepare train dataloader
    batch_size_per_worker = BATCH_SIZE // session.get_world_size()
    train_dataloader = torch.utils.data.DataLoader(
        train_dataset,
        batch_size=batch_size_per_worker
    )
    train_dataloader = ray.train.torch.prepare_data_loader(train_dataloader)
    # prepare optimizer
    optimizer = composer.optim.DecoupledSGDW(
        model.parameters(),
        lr=0.05,
        momentum=0.9,
        weight_decay=2.0e-3,
    )
    return composer.trainer.Trainer(
        model=model,
        train_dataloader=train_dataloader,
        optimizers=optimizer,
        **config
    )
scaling_config = ScalingConfig(num_workers=2, use_gpu=True)
trainer_init_config = {
    "max_duration": "1000ba",
    "algorithms": [LabelSmoothing()],
} 
trainer = MosaicTrainer(
    trainer_init_per_worker=trainer_init_per_worker,
    trainer_init_config=trainer_init_config,
    scaling_config=scaling_config,
) 
trainer.fit() 

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
